synoptic/src/synoptic/model/export/JsonExporter.java

```diff
@@ -96,151 +110,326 @@ public EventInstance(int traceID, int eventIndexWithinTrace) {
-            // One trace, contains the trace number and a list of events
-            Map<String, Object> singleTraceMap = new LinkedHashMap<String, Object>();
-            // List of events
-            List<Map<String, Object>> singleTraceEventsList = new LinkedList<Map<String, Object>>();
-
-            int eventIndexWithinTrace = 0;
-            for (EventNode event = startingEvent; !event
-                    .isTerminal(); event = event.getAllSuccessors().iterator()
-                            .next()) {
-                // One event, contains event index, event type, and timestamp
-                Map<String, Object> singleEventMap = new LinkedHashMap<String, Object>();
+        return nodeList;
+    }
 
-                // Populate this event's index within the trace, its type, and
-                // the line in the log where it was found
-                singleEventMap.put("eventIndex", eventIndexWithinTrace);
-                EventType evType = event.getEType();
-                singleEventMap.put("eventType", evType.toString());
-                singleEventMap.put("logLine", event.getLineNum());
+    // Map of edges and their respective global ID's
+    static Map<EvBasedEdge, Integer> edgesIDMap = new LinkedHashMap<EvBasedEdge, Integer>();
 
-                // Populate this event's time if it's not INITIAL or TERMINAL
-                if (!evType.isSpecialEventType()) {
-                    singleEventMap.put("timestamp", event.getTime());
+    /**
+     * Creates the 'edges' of the JSON object: a list of edges within this
+     * EvBasedGraph
+     * 
+     * @param evGraph
+     *            The EvBasedGraph whose edges we're outputting
+     */
+    private static List<Map<String, Integer>> makeEdgesJSON(
+            EvBasedGraph evGraph) {
+
+        // List of edges and info to go into the JSON Object
+        List<Map<String, Integer>> edgesList = new LinkedList<Map<String, Integer>>();
+
+        // Get all the nodes in the evGraph
+        Set<EvBasedNode> allNodes = evGraph.getNodes();
+
+        // Contains all edges used to avoid duplicates
+        Set<EvBasedEdge> usedEdges = new HashSet<EvBasedEdge>();
+
+        for (EvBasedNode node : allNodes) {
+
+            // Get the edges for each node
+            Set<EvBasedEdge> allEdges = node.outEdges;
+
+            for (EvBasedEdge edge : allEdges) {
+
+                Map<String, Integer> edgesMap = new LinkedHashMap<String, Integer>();
+
+                // Check to see if it is a duplicate
+                if (!usedEdges.contains(edge)) {
+
+                    edgesMap.put("id", globalID);
+                    edgesMap.put("srcNodeID", nodesIDMap.get(edge.srcNode));
+                    edgesMap.put("destNodeID", nodesIDMap.get(edge.destNode));
+                    String displayableValue = edge.eType.getETypeLabel() + " ["
+                            + edge.resMin + ", " + edge.resMax + "]";
+                    if (edge.resMin == null && edge.resMax == null) {
```

Comment:
This means displayableValue will be set twice for every edge for Synoptic models and still twice for a few edges touching initial or terminal in Perfume models. Instead, just assign displayableValue once based on whether this condition is true (so basically do an `if/else` instead of just an `if`).